mpr-research-data.py

```diff
@@ -108,9 +141,89 @@ def retrieveQueryMaker(retrieveQueryTemplate, courseIDs, engine, defaultQueryFol
     return retrieveDF
 
 
-def sliceAndPushToGCP(courseDF, retrieveDF, bucket):
+def updateCourseTimestampTable(courseDF, retrieveDF, client, targetTableID, timestampTableID):
+
+    timestampDFCols = ['CourseID', 'Course',
+                       'CommentCount', 'CourseUploadTime', 'isPredicted']
+
+    timestampQuery = f"SELECT * FROM `{timestampTableID}`"
+    try:
+        timestampDF = client.query(timestampQuery).result().to_dataframe()
+        logging.info(f'Found existing timestamp schema.')
+    except GCPExceptions.NotFound as e:
+        timestampDF = pd.DataFrame(columns=timestampDFCols)
+
+    currentTime = datetime.datetime.now().isoformat()
+    timestampDFRowList = []
+    for _, row in courseDF.iterrows():
+        (courseID, courseName) = row
+        currentLen = len(retrieveDF[retrieveDF['CourseID'] == courseID])
+        logging.info(f'Processing data for {courseID} - {courseName}.')
+
+        if courseID not in timestampDF['CourseID'].values:
+            logging.info(f'{courseName} new to table, adding to GCP.')
+            pushCourseToGCPTable(courseID, retrieveDF,
```

Comment:
since the pushCourseToGCPTable returns T/F values, maybe it should be captured, so that only execute the following timestampDFRowList.append() line when the return value is True.

Reply:
Done.